### PR TITLE
Fix & split-up My VA Payments and Debts E2E spec

### PR DESCRIPTION
**Pull request by tlei123**
## Description

- Fix Cypress-TestRail integration in My VA (Dashboard) Payments and Debts E2E spec.
- Split into smaller, more maintainable specs.

## Original issue
department-of-veterans-affairs/va.gov-team#48978

## Testing done

Locally ran new specs, and results [successfully posted to TestRail](https://dsvavsp.testrail.io/index.php?/runs/overview/4) [see **Saturday, October 29, 2022**].
[See also screenshot below.]

### Screenshot
**TestRail results**:
image

## Acceptance criteria
- [x] New specs also passed in CI-checks below.
- [ ] Code-reviewer(s) approved changes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  describe('when the feature is hidden', () => {
+    beforeEach(() => {
+      cy.intercept('GET', '/v0/feature_toggles*', {
+        data: {
+          type: 'feature_toggles',
+          features: [],
+        },
+      }).as('featuresA');
+    });
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js:53]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            name: featureFlagNames.showPaymentAndDebtSection,
+            value: true,
+          },
+        ],
+      },
+    }).as('featuresD');
+    cy.intercept('/v0/debts', debtsSuccessEmpty()).as('noDebtsD');
+    cy.intercept('/v0/medical_copays', copaysSuccessEmpty()).as('noCopaysD');
+  });
+  // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js:263]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      cy.findByTestId('copay-card-v2').should('exist');
+      cy.findByTestId('copay-due-header-v2').should('exist');
+      cy.findByTestId('manage-va-copays-link-v2').should('exist');
+      cy.findByTestId('learn-va-debt-link-v2').should('exist');
+
+      // make the a11y check
+      cy.injectAxeThenAxeCheck('[data-testid="dashboard-section-debts-v2"]');
+    });
+
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js:86]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      cy.findByTestId('copay-card-v2').should('not.exist');
+      cy.findByTestId('copay-due-header-v2').should('not.exist');
+      cy.findByTestId('no-outstanding-debts-text').should('not.exist');
+      cy.findByTestId('learn-va-debt-link-v2').should('not.exist');
+
+      // make the a11y check
+      cy.injectAxeThenAxeCheck('[data-testid="dashboard-section-debts-v2"]');
+    });
+
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js:128]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    });
+  });
+
+  describe('Debts & Copays API-errors', () => {
+    beforeEach(() => {
+      cy.intercept('/v0/debts', debtsError()).as('debtsErrorB');
+      cy.intercept('/v0/medical_copays', copaysError()).as('copaysErrorB');
+    });
+
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js:153]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2.cypress.spec.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  describe('when the feature is hidden', () => {
+    beforeEach(() => {
+      cy.intercept('GET', '/v0/feature_toggles*', {
+        data: {
+          type: 'feature_toggles',
+          features: [],
+        },
+      }).as('featuresA');
+    });
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2.cypress.spec.js:54]: Previous test already Axe-checked same/similar display-state.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2-error-states.cypress.spec.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import { paymentsError } from '../fixtures/test-payments-response';
+import appointmentsEmpty from '../fixtures/appointments-empty';
+import MOCK_FACILITIES from '../../utils/mocks/appointments/MOCK_FACILITIES.json';
+import { mockLocalStorage } from '~/applications/personalization/dashboard/tests/e2e/dashboard-e2e-helpers';
+import { debtsSuccessEmpty } from '../fixtures/test-debts-response';
+import { copaysSuccessEmpty } from '../fixtures/test-copays-response';
+
+// currently there's just one error-state, but we're keeping blocks-structure for scalability.
+describe('My VA - Benefit payments - error-states', () => {
+  Cypress.config({ defaultCommandTimeout: 12000, requestTimeout: 20000 });
```
Comment (tlei123) [inline src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2-error-states.cypress.spec.js:24]: Also fixed intermittent test-errors from API-request timeouts.

File: src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+  paymentsSuccessEmpty,
+} from '../fixtures/test-payments-response';
+import { debtsSuccessEmpty } from '../fixtures/test-debts-response';
+import { copaysSuccessEmpty } from '../fixtures/test-copays-response';
+import appointmentsEmpty from '../fixtures/appointments-empty';
+import MOCK_FACILITIES from '../../utils/mocks/appointments/MOCK_FACILITIES.json';
+import { mockLocalStorage } from '~/applications/personalization/dashboard/tests/e2e/dashboard-e2e-helpers';
+
+describe('The My VA Dashboard - Benefit Payments', () => {
+  Cypress.config({ defaultCommandTimeout: 12000, requestTimeout: 20000 });
```
Comment (tlei123) [inline src/applications/personalization/dashboard/tests/e2e/benefit-payments-v2.cypress.spec.js:28]: Also fixed intermittent test-errors from API-request timeouts.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+  copaysError,
+  copaysSuccess,
+  copaysSuccessEmpty,
+} from '../fixtures/test-copays-response';
+import appointmentsEmpty from '../fixtures/appointments-empty';
+import MOCK_FACILITIES from '../../utils/mocks/appointments/MOCK_FACILITIES.json';
+import { mockLocalStorage } from '~/applications/personalization/dashboard/tests/e2e/dashboard-e2e-helpers';
+
+describe('My VA - Outstanding debts error-states', () => {
+  Cypress.config({ defaultCommandTimeout: 12000, requestTimeout: 20000 });
```
Comment (tlei123) [inline src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js:31]: Also fixed intermittent test-errors from API-request timeouts.

File: src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2.cypress.spec.js [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+import {
+  copaysSuccess,
+  copaysSuccessEmpty,
+} from '../fixtures/test-copays-response';
+import appointmentsEmpty from '../fixtures/appointments-empty';
+import MOCK_FACILITIES from '../../utils/mocks/appointments/MOCK_FACILITIES.json';
+import { mockLocalStorage } from '~/applications/personalization/dashboard/tests/e2e/dashboard-e2e-helpers';
+
+describe('The My VA Dashboard - Outstanding Debts', () => {
+  Cypress.config({ defaultCommandTimeout: 12000, requestTimeout: 25000 });
```
Comment (tlei123) [inline src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2.cypress.spec.js:29]: Also fixed intermittent test-errors from API-request timeouts.

**Comment by tlei123** [comment]
@tmitchellgcio , please review & approve this PR.

All newly-created [split-up] specs passed in CI.  Results now successfully posted to TestRail [see main description above].

Thanks much in advance!  :)

**Review by it-harrison — APPROVED** [review]

**Review by tmitchellgcio — APPROVED** [review]
LGTM

**Pull request merged**